### DISCUSSION TITLE
config.json

### DISCUSSION THREAD

**Discussion by jt1971** [tsightler/ring-mqtt#456] [Support Q&A] [answered]
I have tired to source information regarding this but I am a little stuck and would value a nudge in the right direction.

I am running HA OS on a virtual box on a Windows PC.

Installed the Add On successfully, authenicated okay with my Ring credentials.

The readme and wiki refer to updating the config.json.  Where is it, and how do I get to it - I cannot find it from the File Editor.

Have Mosquito up and running on a Raspberry Pi which has been stable for a number of years serving other home automations - I am trying to bring everything together on HA.

Thanks in advance.

**Comment by tsightler** [comment]
For the addon you should not need to mess with any configuration files, all config options are managed from the addons configuration tab.

Just for completeness the config.json file is actually named options.json and can be accessed from within the addon container in the /data/options.json path, however, you should never actually need to do this.

**Comment by jt1971** [comment]
Thanks for such a quick reply.  What sent me down that direction was the comment in the readme 'Prior to running the container for the first time you should edit the <host_path>/config.json file with the appropriate information.'

Where do I enter my mqtt broker IP address for example. Thanks again.

**Reply by tsightler** [reply]
Hmm...I'm amazed nobody every pointed that out before, looks like a copy/paste error when the docs were moved to Wiki format, I'll fix it.

Regardless, the link to the global configuration options is still correct and aligns with the options in the addon configuration.  MQTT connection settings are defined via the mqtt_url config options as documented.

**Reply by jt1971** [reply]
Thanks for your patience.  It looks like a great add on, but I'm still stuck where to update the MQTT connection settings.
If I click on web UI from within the Add On, it just asks about regenerating my ring API token.  I appreciate I must be missing something somewhere.  Thanks again.

**Reply by tsightler** [reply]
It's not the web UI of the addon, the configuration is done in the normal Home Assistant UI, just click the configuration tab and all config options are there, just like all other addons.

**Reply by jt1971** [reply]
I'm sorry to come back again, but perhaps we are talking about different versions of HA?  Mine is:
Home Assistant 2022.9.0
Supervisor 2022.08.6
Operating System 8.5
Frontend 20220907.0 - latest

There is no configuration tab; just a settings button on the bottom left.  This brings up various items such as 'devices & services', 'automations & Scenes'....'add on's'. In the latter is the ring-mqtt but no configuration options there or anywhere else in the settings that I can find.  

the only entity relating to your plugin that is populated is 'update.ring_mqtt_with_video_streaming_update' which refers to a warning re. 5.0.3 release and compatibility with previous versions which I don't think is relevant as this is a fresh install.

**Comment by jeffgraham-hub** [comment, marked as answer]
@jt1971 There should be a configuration option at the top after selecting add ons/ring-mqtt.


![image](https://user-images.githubusercontent.com/65316307/189165916-d4ab4674-e601-4a25-b705-5a21bc81307f.png)

**Reply by jt1971** [reply]
I'm sorry. I knew it would be obvious - I just didn't see that for looking!  Thanks a million.

**Reply by tsightler** [reply]
Thanks for jumping in @jeffgraham-hub.  I guess it seemed so obvious that it didn't cross my mind that it could be missed, but you know what, now looking at it from the UI perspective, it's not really 100% obvious that those are basically tabs.  I'll try to fix the docs in the wiki and maybe include a screenshot just to be 100% sure for future users.
